Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -601,6 +601,38 @@ public static Branch getAutoInstance(@NonNull Context context, boolean isReferra
     }
 
     /**
+     * <p>Singleton method to return the pre-initialised, or newly initialise and return, a singleton
+     * object of the type {@link Branch}.</p>
+     *
+     * <p>Use this whenever you need to call a method directly on the {@link Branch} object.</p>
+     *
+     * @param context A {@link Context} from which this call was made.
+     * @param branchKey A {@link String} value used to initialize Branch.
+     * @return An initialised {@link Branch} object, either fetched from a pre-initialised
+     * instance within the singleton class, or a newly instantiated object where
+     * one was not already requested during the current app lifecycle.
+     */
+    @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH)
+    public static Branch getAutoInstance(@NonNull Context context, String branchKey) {
+        isAutoSessionMode_ = true;
+        customReferrableSettings_ = CUSTOM_REFERRABLE_SETTINGS.USE_DEFAULT;
+        boolean isLive = !BranchUtil.isTestModeEnabled(context);
+        getBranchInstance(context, isLive);
+
+        if (branchKey.startsWith("key_")) {
```

Comment:
This context should be an Application instance. I think it would be better to specify that in the comment